buildSrc/src/main/kotlin/otel.muzzle-check.gradle.kts

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import io.opentelemetry.instrumentation.gradle.muzzle.AcceptableVersions
+import io.opentelemetry.instrumentation.gradle.muzzle.BogusClassLoader
+import io.opentelemetry.instrumentation.gradle.muzzle.MuzzleDirective
+import io.opentelemetry.instrumentation.gradle.muzzle.MuzzleExtension
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils
+import org.eclipse.aether.RepositorySystem
+import org.eclipse.aether.RepositorySystemSession
+import org.eclipse.aether.artifact.Artifact
+import org.eclipse.aether.artifact.DefaultArtifact
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory
+import org.eclipse.aether.repository.LocalRepository
+import org.eclipse.aether.repository.RemoteRepository
+import org.eclipse.aether.resolution.VersionRangeRequest
+import org.eclipse.aether.resolution.VersionRangeResult
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory
+import org.eclipse.aether.spi.connector.transport.TransporterFactory
+import org.eclipse.aether.transport.http.HttpTransporterFactory
+import org.eclipse.aether.version.Version
+import java.net.URL
+import java.net.URLClassLoader
+import java.util.stream.StreamSupport
+
+plugins {
+  `java-library`
+}
+
+// Select a random set of versions to test
+val RANGE_COUNT_LIMIT = 10
+
+val muzzleConfig = extensions.create<MuzzleExtension>("muzzle")
+
+val muzzleTooling by configurations.creating {
```

Comment:
Creating the configurations in the plugin instead of the project like before as plugins normally do. I haven't quite been able to reason out the difference between these configurations yet (intuitively I'd expect us to only need one classpath when muzzling) but have reproduced the split for now.

Reply:
I think one of them (`muzzleTooling`) is supposed to be used as the agent classpath source (to be able to load `InstrumentationModule`), the other (`muzzleBootstrap`) should be a base for the "instrumented application" classpath - add the library jar for a given version to our bootstrap classes and pass it to the check method.

Reply:
Ah thanks that makes sense!